packages/plugin-kafka/src/kafka.ts

```diff
@@ -0,0 +1,151 @@
+import assert = require('assert')
+import { resolve } from 'path'
+import { NotFoundError } from 'common-errors'
+import { Microfleet, PluginTypes, LoggerPlugin } from '@microfleet/core'
+import debugLog from 'debug'
+import { promisifyAll, map } from 'bluebird'
+
+import {
+  KafkaConsumer,
+  Producer as KafkaProducer,
+  ProducerStream,
+  ConsumerStream,
+} from 'node-rdkafka'
+
+promisifyAll(KafkaConsumer.prototype)
+promisifyAll(KafkaProducer.prototype)
+
+import {
+  TopicConfig,
+  ProducerStreamOptions,
+  ConsumerStreamOptions,
+  KafkaConfig,
+} from './types'
+
+const debug = debugLog('plugin-kafka')
+
+/**
+ * Library hides consumer whe using typescript
+ * https://blizzard.github.io/node-rdkafka/current/KafkaConsumerStream.html
+ * But direct consumer creation was requested in
+ * https://github.com/microfleet/core/pull/362#discussion_r367773758
+ */
+const kConsumerStream = require('node-rdkafka/lib/kafka-consumer-stream')
+const kProducerStream = require('node-rdkafka/lib/producer-stream')
+
+promisifyAll(kConsumerStream.prototype)
+promisifyAll(kProducerStream.prototype)
+
+/**
+ * Relative priority inside the same plugin group type
+ */
+export const priority = 0
+export const name = 'kafka'
+export const type = PluginTypes.transport
+
+/**
+ * Defines service extension
+ */
+export interface KafkaPlugin {
+  rdKafkaConfig: KafkaConfig
+  createConsumerStream: (
+    streamOptions: ConsumerStreamOptions,
+    conf?: KafkaConfig,
+    topicConf?: TopicConfig
+  ) => Promise<ConsumerStream>
+  createProducerStream: (
+    streamOptions: ProducerStreamOptions,
+    conf?: KafkaConfig,
+    topicConf?: TopicConfig
+  ) => Promise<ProducerStream>
+  close: () => Promise<void>
+}
+
+type AnyStream = ConsumerStream | ProducerStream
+
+export class KafkaFactory implements KafkaPlugin {
+  rdKafkaConfig: KafkaConfig
+  private _streams: Set<AnyStream>
+
+  constructor(config: KafkaConfig) {
+    this.rdKafkaConfig = config
+    this._streams = new Set<AnyStream>()
+  }
+
+  async connectStream(stream: AnyStream) {
+    const client = Object.prototype.hasOwnProperty.call(stream, 'consumer')
+    ? (stream as ConsumerStream).consumer
+    : (stream as ProducerStream).producer
+
+    // Kafka debugging
+    client.on('event.log', (log: any) => {
+      debug(`[${log.severity}](${log.fac}): ${log.message}`)
```

Comment:
Use standard logger plugin, as this is useful information. Log level should be chosen based on severity (mapping between kafka severity and logger needs to be created)